File: IT/par61/task5.py
def bin_dec(N, k=0, R=0):
    ts = N % 10
    R += ts * 2 ** k
    if N > 1 :
        return bin_dec(N // 10, k + 1, R)
    return R


number = int(input("напиши число из 10 счисление "))
print(f'число {number} переводим в десятичную счисление: {bin_dec(number)}')

File: php/php4/task5.php
<?php
if ($_SERVER['REQUEST_METHOD'] === 'POST') {
    $number = $_POST['number'];

    if (ctype_digit($number)) {
        $reversedNumber = strrev($number);
        echo "Число задом наперед: $reversedNumber";
    } else {
        echo "Ошибка: введены нечисловые символы.";
    }
}
?>

<!DOCTYPE html>
<html lang="ru">
<head>
    <meta charset="UTF-8">
    <title>Переворот числа</title>
</head>
<body>
    <form method="POST" action="">
        <label for="number">Введите число:</label>
        <input type="text" id="number" name="number" required>
        <button type="submit">Отправить</button>
    </form>
</body>
</html>